src/world/components/lib/checkTransitionMatch.js

```diff
@@ -0,0 +1,14 @@
+const _ = require('lodash');
+const Type = require('../../../components/type.js');
+
+module.exports = function checkTransitionMatch(rule, thing, locations, action) {
+  const ruleSource = rule.getSource();
+  const targetLocation = rule.getConsequentTarget();
+  const sourceMatch = ruleSource instanceof Type
+    ? _.contains(thing.getTypes(), ruleSource.get())
+    : ruleSource === thing.id;
+  const originMatch = thing.location === rule.getTarget();
+  const destinationMatch = _.contains(locations, targetLocation);
```

Comment:
This makes me wish we had Node 6 in play so we could use [`[].includes`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes).